lib/config/index.js

```diff
@@ -63,75 +58,75 @@ if (config.ldap.tlsca) {
   let tlsOptions = {
     ca: caContent
   }
-  config.ldap.tlsOptions = config.ldap.tlsOptions ? Object.assign(config.ldap.tlsOptions, tlsOptions) : tlsOptions
+  config.load({
+    ldap: {
+      tlsOptions: config.get(ldap).tlsOptions ? Object.assign(config.get('ldap').tlsOptions, tlsOptions) : tlsOptions
+    }
+  })
 }
 
 // Permission
-config.permission = Permission
+let permissions = Permission
 if (!config.allowAnonymous && !config.allowAnonymousEdits) {
-  delete config.permission.freely
+  delete permissions.freely
 }
-if (!(config.defaultPermission in config.permission)) {
-  config.defaultPermission = config.permission.editable
+if (!(config.defaultPermission in permissions)) {
+  config.load({defaultPermission: permissions.editable})
 }
 
-// cache result, cannot change config in runtime!!!
-config.isStandardHTTPsPort = (function isStandardHTTPsPort () {
-  return config.useSSL && config.port === 443
-})()
-config.isStandardHTTPPort = (function isStandardHTTPPort () {
-  return !config.useSSL && config.port === 80
-})()
+config.load({permission: permissions})
 
 // cache serverURL
-config.serverURL = (function getserverurl () {
-  var url = ''
-  if (config.domain) {
-    var protocol = config.protocolUseSSL ? 'https://' : 'http://'
-    url = protocol + config.domain
-    if (config.urlAddPort) {
-      if (!config.isStandardHTTPPort || !config.isStandardHTTPsPort) {
-        url += ':' + config.port
+config.load({
+  serverURL: (function getserverurl () {
+    const isStandardHTTPsPort = config.get('useSSL') && config.get('port') === 443
+    const isStandardHTTPPort = config.get('useSSL') && config.get('port') === 80
```

Comment:
the second one should probably be `!config.get('useSSL')` (note the "not" in front)